src/hex_api_auth.erl

```diff
@@ -0,0 +1,26 @@
+-module(hex_api_auth).
+-export([
+    test_key/3
+]).
+
+-define(ERL_CONTENT_TYPE, <<"application/vnd.hex+erlang">>).
+
+%% @doc
+%% Test an auth key against a given domain and resource.
+%%
+%% Examples:
+%%
+%% ```
+%% 1> Params = [{domain, <<"repository">>}, {resource, <<"gustafson_motors">>}].
+%% [{domain,<<"repository">>},
+%% {resource,<<"gustafson_motors">>}]
+%% 2> Key = <<"86c4ac12292843b66fdee41e022f8497">>
```

Comment:
We shouldn't pass the key explicitly, instead we should use the key from the config.